### PR TITLE
YAML: create yaml testing files for subroutines in modal_aer_opt

### PR DESCRIPTION
**Pull request by tangshuaiqi**
creating yaml files for subroutines in modal_aer_opt.F90. Some module variables and diagnostics variables are included besides the in/out variables. There are a few 3-d or 5-d arrays, which are reshaped into a 1-d array and written out. The FORTRAN buildin reshape function has a default natural order of permutation from 1 to n.

### CODE REVIEW

**Review by jeff-cohere — APPROVED** [review]
Thanks!

**Review by singhbalwinder — APPROVED** [review]
Thanks, @tangshuaiqi !

File: components/eam/src/chemistry/yaml/modal_aer_opt/f90_yaml/calc_refin_complex_beg_yml.f90 [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+             'calc_refin_complex',yaml%nstep_print, yaml%lev_print)
+
+        ! add code for writing data here
+        
+        call write_var(unit_input,unit_output,'ncol',ncol)
+        call write_var(unit_input,unit_output,'ilwsw',ilwsw)
+        call write_var(unit_input,unit_output,'qaerwat_kk',qaerwat_kk(yaml%col_print))
+        call write_var(unit_input,unit_output,'specvol',specvol(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'specrefindex',specrefindex(yaml%col_print,:))
+
```
Comment (odiazib) [inline components/eam/src/chemistry/yaml/modal_aer_opt/f90_yaml/calc_refin_complex_beg_yml.f90]: please add  [crefwlw](https://github.com/eagles-project/e3sm_mam4_refactor/blob/966023e8429c1f913df38743dc84a1ab9911406b/components/eam/src/physics/cam/modal_aer_opt.F90#L1201C11-L1201C11) and [crefwsw](https://github.com/eagles-project/e3sm_mam4_refactor/blob/966023e8429c1f913df38743dc84a1ab9911406b/components/eam/src/physics/cam/modal_aer_opt.F90#L1204C56-L1204C63)

File: components/eam/src/chemistry/yaml/modal_aer_opt/f90_yaml/modal_aero_lw_beg_yml.f90 [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        ! add code for writing data here
+        
+        call write_var(unit_input,unit_output,'dt',dt)
+        call write_var(unit_input,unit_output,'ncol',ncol)
+        call write_var(unit_input,unit_output,'state_q',state_q(yaml%col_print,:,:))
+        call write_var(unit_input,unit_output,'temperature',temperature(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'pmid',pmid(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'pdel',pdel(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'pdeldry',pdeldry(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'cldn',cldn(yaml%col_print,:))
```
Comment (odiazib) [inline components/eam/src/chemistry/yaml/modal_aer_opt/f90_yaml/modal_aero_lw_beg_yml.f90:100]: please add: `refrtablw`, `refitablw`, and `specrefndxlw`; [see here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/966023e8429c1f913df38743dc84a1ab9911406b/components/eam/src/physics/cam/modal_aer_opt.F90#L32)

**Review by odiazib — CHANGES_REQUESTED** [review]
It looks great! I just asked to add a few variables.

**Comment by tangshuaiqi** [comment]
@odiazib Done. and added a few more module variables that may be needed in the subroutines.

**Review by odiazib — APPROVED** [review]
Thanks!

**Comment by tangshuaiqi** [comment]
all tests passed. merge into maint

comp_testall_yaml_opt_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**